telegram/ext/filters.py

```diff
@@ -372,1895 +363,1807 @@ def name(self, name: str) -> NoReturn:
-            """
-            return super().add_chat_ids(chat_id)
-
-        def remove_usernames(self, username: SLT[str]) -> None:
-            """
-            Remove one or more sender chats from allowed usernames.
-
-            Args:
-                username(:obj:`str` | Tuple[:obj:`str`] | List[:obj:`str`], optional):
-                    Which sender chat username(s) to disallow through.
-                    Leading ``'@'`` s in usernames will be discarded.
-            """
-            return super().remove_usernames(username)
-
-        def remove_chat_ids(self, chat_id: SLT[int]) -> None:
-            """
-            Remove one or more sender chats from allowed chat ids.
-
-            Args:
-                chat_id(:obj:`int` | Tuple[:obj:`int`] | List[:obj:`int`], optional):
-                    Which sender chat ID(s) to disallow through.
-            """
-            return super().remove_chat_ids(chat_id)
-
-        class _SuperGroup(MessageFilter):
-            __slots__ = ()
-
-            def filter(self, message: Message) -> bool:
-                if message.sender_chat:
-                    return message.sender_chat.type == Chat.SUPERGROUP
-                return False
+        def filter(self, message: Message) -> bool:
+            return message.chat.type == TGChat.GROUP
 
-        class _Channel(MessageFilter):
-            __slots__ = ()
+    GROUP = _Group()
+    """Updates from group."""
 
-            def filter(self, message: Message) -> bool:
-                if message.sender_chat:
-                    return message.sender_chat.type == Chat.CHANNEL
-                return False
+    class _SuperGroup(MessageFilter):
+        __slots__ = ()
+        name = 'filters.ChatType.SUPERGROUP'
+
+        def filter(self, message: Message) -> bool:
+            return message.chat.type == TGChat.SUPERGROUP
 
-        super_group = _SuperGroup()
-        channel = _Channel()
+    SUPERGROUP = _SuperGroup()
+    """Updates from supergroup."""
 
-    class _Invoice(MessageFilter):
+    class _Groups(MessageFilter):
         __slots__ = ()
-        name = 'Filters.invoice'
+        name = 'filters.ChatType.GROUPS'
 
         def filter(self, message: Message) -> bool:
-            return bool(message.invoice)
+            return message.chat.type in [TGChat.GROUP, TGChat.SUPERGROUP]
 
-    invoice = _Invoice()
-    """Messages that contain :class:`telegram.Invoice`."""
+    GROUPS = _Groups()
+    """Update from group *or* supergroup."""
 
-    class _SuccessfulPayment(MessageFilter):
+    class _Private(MessageFilter):
         __slots__ = ()
-        name = 'Filters.successful_payment'
+        name = 'filters.ChatType.PRIVATE'
 
         def filter(self, message: Message) -> bool:
-            return bool(message.successful_payment)
+            return message.chat.type == TGChat.PRIVATE
+
+    PRIVATE = _Private()
+    """Update from private chats."""
+
+
+class _ChatUserBaseFilter(MessageFilter, ABC):
```

Comment:
The properties - we provide a way for the user to see the current chat_ids/usernames, so why should it become private?

Reply:
E.g. `ViaBot` exposes the `bot_ids` property - it's just a different naming but it makes more sense in the context. Same for `User`. Okay, `usernames`  can probably stay as is, because that's the same naming everywhere …

Also I just notice that [`get_chat_or_user`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.ext.filters.html#telegram.ext.filters.Filters.user.get_chat_or_user) appears in the docs for some reason. we should make that private as well …

Reply:
> Also I just notice that [`get_chat_or_user`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.ext.filters.html#telegram.ext.filters.Filters.user.get_chat_or_user) appears in the docs for some reason. we should make that private as well …

It's already fixed on the v14 branch, so issue not present here.